docs/design/seccomp.md

```diff
@@ -0,0 +1,364 @@
+specification of seccomp filters, with arguments, operators, and actions.
+
+Docker allows the specification of a single seccomp filter.  There are
+community requests for:
+
+Issues:
+
+* [docker/22109](https://github.com/docker/docker/issues/22109): composable
+  seccomp filters
+* [docker/21105](https://github.com/docker/docker/issues/22109): custom
+  seccomp filters for builds
+
+#### rkt / appcontainers
+
+The `rkt` runtime delegates to systemd for seccomp support; there is an open
+issue to add support once `appc` supports it.  The `appc` project has an open
+issue to be able to describe seccomp as an isolator in an appc pod.
+
+The systemd seccomp facility is based on a whitelist of system calls that can
+be made, rather than a full filter specification.
+
+Issues:
+
+* [appc/529](https://github.com/appc/spec/issues/529)
+* [rkt/1614](https://github.com/coreos/rkt/issues/1614)
+
+#### HyperContainer
+
+[Hyper](https://hyper.sh) uses a hypervisor to run containers, so seccomp is
+not an applicable concept.
+
+### Other platforms and seccomp-like capabilities
+
+FreeBSD has a seccomp/capability-like facility called
+[Capsicum](https://www.freebsd.org/cgi/man.cgi?query=capsicum&sektion=4).
+
+#### lxd
+
+[`lxd`](http://www.ubuntu.com/cloud/lxd) constrains containers using a default profile.
+
+Issues:
+
+* [lxd/1084](https://github.com/lxc/lxd/issues/1084): add knobs for seccomp
+
+## Proposed Design
+
+### Seccomp API Resource
+
+We should have an API resource to describe seccomp profiles.  Since docker and
+runc both support a rich seccomp API, in order to capture default profiles,
+our own API for seccomp should be close to the libseccomp API as well.
+
+Making this API resource non-namespaced has several advantages:
+
+1.  It fits with the design of seccomp profiles as administrator-governed
+2.  It makes it easier to implement an access policy
+
+```go
+
+// Represents a seccomp profile.  A profile is composed of a list of seccomp
+// rules and a default action.  Seccomp profiles are unnamespaced.
+type SeccompProfile struct {
+	unversioned.TypeMeta `json:",inline"`
+	ObjectMeta           `json:"metadata,omitempty"`
+
+	Rules         []SeccompRule `json:"syscalls,omitempty"`
+	DefaultAction SeccompAction `json:"defaultAction"`
+}
+
+// A seccomp syscall filter rule.
+type SeccompRule struct {
+	// The name of the system call to use
+	Name   string `json:"name"`
+	// The action to take
+	Action Action `json:"action"`
+	// The arguments to the filter
+	Args   []*Arg `json:"args"`
+}
+
+// An action associated with a seccomp rule.
+type SeccompAction string
+
+const (
+	Kill  SeccompAction = "kill"
+	Errno SeccompAction = "errno"
+	Trap  SeccompAction = "trap"
+	Allow SeccompAction = "allow"
+	Trace SeccompAction = "trace"
+)
+
+// A comparison operator to be used when matching syscall arguments in Seccomp
+type SeccompOperator string
+
+const (
+	EqualTo              SeccompOperator = "equalTo"
```

Comment:
API constant convention is to start with a capital letter.

Also, please be consistent with selector operators where there is overlap.